### PR TITLE
Fix latex repr of symbolic distributions

### LINKED ISSUE #6202: Improve Latex representation of symbolic distributions
Symbolic distributions often (or *always(?)*) use other distributions as input. As such, when displaying the data-generating model in Latex, it does not look properly rendered, although we can understand what the model is. Here are two examples:

```python
with pm.Model() as censored_model:
    mu = pm.Normal("mu", 0., 5.)
    sigma = pm.HalfCauchy("sigma", 2.5)
    normal_dist = pm.Normal.dist(mu=mu, sigma=sigma)
    censored_normal = pm.Censored("censored_normal", normal_dist, lower=-2., upper=2., observed=[1, 0, 0.5])
```
yields
Screen Shot 2022-10-11 at 9 32 16 AM

and

```python
with pm.Model() as mix_model:
    w = pm.Dirichlet("w", [1, 1])
    mix = pm.Mixture("mix", w=w, comp_dists=[pm.Normal.dist(0., 5.), pm.StudentT.dist(7.)])
```

yields
Screen Shot 2022-10-11 at 9 33 08 AM

### PR DESCRIPTION
**Pull request by mattiadg**
Fix #6202 

* Do not put \operatorname inside \text
* add test

**What is this PR about?**
When a symbolic distribution is defined as parameter of another one, the latex representation is broken.
This PR aims to solve this.

I think that the new failing tests are testing the wrong behavior, this is a draft because I would like someone more experienced with this to check on them. 
What I'm doing is to remove `\operatorname` from inside `\text` because it's wrong.

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- Fix broken latex representation of distributions inside other distributions

## Docs / Maintenance
- ...

### CODE REVIEW

**Comment by mattiadg** [comment]
Is anything else needed here?

**Comment by mattiadg** [comment]
I've already solved conflicts twice because people keep adding wrong tests for the latex printing.
That functionality is currently broken as #6202 shows, so the tests must be wrong in order to fail.
Please, have a loot at this PR before merging additional code in test_printing.py.

**Comment by ricardoV94** [comment]
> I've already solved conflicts twice because people keep adding wrong tests for the latex printing. That functionality is currently broken as #6202 shows, so the tests must be wrong in order to fail. Please, have a loot at this PR before merging additional code in test_printing.py.

That sometimes happens, a reviewer may be able to get to another PR and merge it before yours. Sorry for the trouble, but it's also something one must get used to :)

**Comment by mattiadg** [comment]
Yes, I understand that, I don't want to look upset. I'm just saying that
the current tests in that file, and the ones are being added to it, are
(partially) testing the wrong behavior, which is the reason why that issue
exists in the first place.

File: pymc/printing.py [hunk 1]
```diff
@@ -215,9 +217,20 @@ def _expand(x):
...
+    else:
+        return r"f(" + ", ".join([n.strip("$") for n in names]) + ")"
+
+
+def _latex_text_format(text: str) -> str:
+    if "operatorname" in text:
+        return text
     else:
-        return r"f(" + ", ".join(names) + ")"
+        return r"\text{" + text + "}"
```
Comment (Spaak) [inline pymc/printing.py:233]: Many lines are now flagged as not covered by tests. But it seems to me that your `test_model_latex_repr_nested` should have tested these, correct? Perhaps that test is not triggered somehow. Would be good to fix before merging.

File: pymc/printing.py [hunk 2]
```diff
@@ -215,9 +217,20 @@ def _expand(x):
...
+            r"f("
+            + ",~".join([_latex_text_format(_latex_escape(n.strip("$"))) for n in names])
+            + ")"
+        )
+    else:
+        return r"f(" + ", ".join([n.strip("$") for n in names]) + ")"
+
+
+def _latex_text_format(text: str) -> str:
+    if "operatorname" in text:
```
Comment (Spaak) [inline pymc/printing.py]:
```suggestion
    if "\\operatorname{" in text:
```

**Review by Spaak — CHANGES_REQUESTED** [review]
LGTM in general, though the testing should be fixed.

File: pymc/printing.py (hunk 2 above)
Comment (mattiadg) [reply pymc/printing.py]: I pushed this change and fixed the failing test. One reference value was wrong after merging.
Now, I check why the coverage goes down.

**Review by Spaak — APPROVED** [review]
LGTM!

**Comment by Spaak** [comment]
Looks good to me, thanks for adding the tests so quickly. I'll leave merging to someone else (eg @ricardoV94) because it's been a while for me.

**Comment by mattiadg** [comment]
> Looks good to me, thanks for adding the tests so quickly. I'll leave merging to someone else (eg @ricardoV94) because it's been a while for me.

That was easy, I added as tests the exact same distributions used in the issue to show the problem. This way it's also possible to reference back to them, should something go wrong in the future.
However, I still need to understand the two branches of Codecov: `patch` and `project`. I am still not sure why `patch` became lower initially

File: pymc/tests/test_printing.py [hunk 3]
```diff
@@ -178,3 +178,43 @@ def test_str_repr(self):
...
+    expected = [
+        "$$",
+        "\\begin{array}{rcl}",
+        "\\text{w} &\\sim & "
+        "\\operatorname{Dirichlet}(\\text{<constant>})\\\\\\text{mix} &\\sim & "
+        "\\operatorname{MarginalMixture}(\\text{w},~\\operatorname{Normal}(0,~5),~\\operatorname{StudentT}(7,~0,~1))",
+        "\\end{array}",
+        "$$",
+    ]
+    assert [line.strip() for line in latex_repr.split("\n")] == expected
```
Comment (ricardoV94) [inline pymc/tests/test_printing.py]: Does this check do anything that the corrected pre-existing checks do not?

File: pymc/printing.py [hunk 4]
```diff
@@ -215,9 +217,20 @@ def _expand(x):
...
+        return (
+            r"f("
+            + ",~".join([_latex_text_format(_latex_escape(n.strip("$"))) for n in names])
+            + ")"
+        )
+    else:
+        return r"f(" + ", ".join([n.strip("$") for n in names]) + ")"
+
+
+def _latex_text_format(text: str) -> str:
```
Comment (larryshamalama) [inline pymc/printing.py:229]: Okay for now, but I have a sense that this can be simplified since you're always using it on an output of `_latex_escape`

**Review by larryshamalama — APPROVED** [review]
Looks good, thanks for your help! Glad that you were able to solve the other conflicts following the merge of #6242

File: pymc/tests/test_printing.py (hunk 3 above)
Comment (mattiadg) [reply pymc/tests/test_printing.py]: They test additionally `HalfCauchy`, `StudentT` and `Model` itself, but I also added these two tests is to keep under track the models reported in the issue.

File: pymc/printing.py (hunk 4 above)
Comment (mattiadg) [reply pymc/printing.py:229]: I can add the call to `_latex_escape(text.strip("$")` inside `_latex_text_format` and call this also in the other 2 occurrences of `_latex_escape`. Then, we can also remove the naked `\text{` in there.

**Comment by michaelosthege** [comment]
@mattiadg there appears to be one more conflict?

Would be great to push this over the finish line as we have a release coming up!

**Comment by mattiadg** [comment]
Sure, I will do it this evening or tomorrow

**Comment by mattiadg** [comment]
> @mattiadg there appears to be one more conflict?
> 
> Would be great to push this over the finish line as we have a release coming up!

done

**Review by michaelosthege — APPROVED** [review]

**Comment by michaelosthege** [comment]
Nice, thank you for fixing these @mattiadg !

**Pull request merged**